pepsico/monthly_seasonal_conv_mapping.py

```diff
@@ -0,0 +1,199 @@
+import numpy as np
+import pandas as pd
+import xarray as xr
+import plotly.express as px
+from plotly.io import write_html
+
+def unit_conversion(ds, variable):
+     #if precipitation variable, change from kg to mm per day
+    if variable == 'pr':
+        ds[variable] *= 86400
+        ds[variable].attrs['units'] = 'mm/day' #rename unit to converted
+    elif variable in ['tas', 'tasmin', 'tasmax']:
+        ds[variable] -= 273.15 
+        ds[variable].attrs['units'] = 'Celsius'
+    
+    return ds 
+
+def compute_monthly_avg(ds, variable):
+    # Convert daily data to monthly
+    monthly_ds = ds.resample(T="1M").mean()
+
+    # spatial conversion
+    monthly_avg = monthly_ds.mean(dim=["X", "Y"])
+
+    return monthly_avg
+
+
+def compute_annual_monthly_avg(monthly_avg, variable):
+    #compute the avg (variable) for each month across the specified years
+
+    #Compute the average for each month over time
+    annual_monthly_avg = monthly_avg.groupby("T.month").mean(dim="T")
+    return annual_monthly_avg
+
+
+def compute_seasonal_avg(monthly_avg):
+    #compute seasonal averages
+    rolling_avg = monthly_avg.rolling(T=3, center=True).mean()
+    return rolling_avg
+    
+def compute_annual_seasonal_avg(rolling_avg, variable):
+    
+    # Compute the average variable for each rolling season across all years
+    rolling_seasonal_avg = rolling_avg.groupby("T.month").mean(dim="T")
+    return rolling_seasonal_avg
+
+
+def create_seasonal_dataframe(rolling_seasonal_avg):
+    # Create a DataFrame for seasonal averages
+    seasonal_avgs_df = rolling_seasonal_avg.to_dataframe().reset_index()
+
+    # Assign the season column with correct length
+    seasonal_avgs_df['season'] = ['DJF', 'JFM', 'FMA', 'MAM', 'AMJ', 'MJJ',
+                                   'JJA', 'JAS', 'ASO', 'SON', 'OND', 'NDJ']
```

Comment:
I think DJF is likely last, and not first. Since you do a rolling of 3 that is centered, and since data always starts in January (whether you select years or not), the first center will be Feb, and so the first season would be JFM and so on and so forth, the last one would be DJF. Unless groupby reorders them? Please double check.